src/server.ts

```diff
@@ -91,39 +89,71 @@ setInterval(() => {
 */
 
 const didConfig = '.well-known/did-configuration.json';
+
+type TimeSinceEpoch = number
+type DateTimeString = string
+type LongFormDID = string
+
+type DomainLinkageCredentialJWTPayload = {
+   exp: TimeSinceEpoch,
+   iss: LongFormDID 
+   nbf: TimeSinceEpoch,
+   sub: LongFormDID,
+   vc: {
+       "@context": [
+           "https://www.w3.org/2018/credentials/v1",
+           "https://identity.foundation/.well-known/did-configuration/v1"
+        ],
+        credentialSubject: {
+            id: LongFormDID,
+            origin: string
+        },
+        expirationDate: DateTimeString,
+        issuanceDate: DateTimeString,
+        issuer: LongFormDID,
+        type: ["VerifiableCredential", "DomainLinkageCredential"]
+   }
+}
+
+type WellKnownDidConfigurationResponse = {
+    "@context": "https://identity.foundation/.well-known/did-configuration/v1",
+    linked_dids: [string]
```

Comment:
I was happy creating types from [the spec](https://identity.foundation/.well-known/resources/did-configuration/#json-web-token-proof-format), but it seems like the JWS signing process breaks the chain of types :/ – any thoughts here? Happy to wind back the types if provides less value than its costs.

Reply:
I think the types are helpful! (To be clear: I wrote this demo while learning typescript, so there's still a bunch of any-typed stuff floating around the project and worth cleaning up over time; love the idea of trying to keep newer contributions better typed ;-))

If you want you could follow the approach from https://stackoverflow.com/a/37074697 so that `linked_dids` has a type like `[SignedDomainLinkageCredentialJWTPayload]`; that's be purely a market (to say, beyond string that is also has this extra JWT-like nature), but you could later define decoder functions that take in a `SignedDomainLinkageCredentialJWTPayload` and return (after validating) a `DomainLinkageCredentialJWTPayload`.

More broadly, you could explore defining a `SignedJWTPayload<T>` type parameterized on the particular payload structure.

Reply:
Good pointers, I'll try a "tagged string" approach.